### PR TITLE
Gzip compression post put

### PR DESCRIPTION
**Pull request by aseure**

### CODE REVIEW

File: algolia/transport/transport.go [hunk 1]
```diff
@@ -193,7 +224,21 @@ func buildRequest(
...
+		var r io.ReadCloser
+
+		if shouldCompress(c, method) {
+			switch c {
+			case compression.GZIP:
+				r = gzipCompressionReadCloser(data)
+			default:
+				r = noCompressionReadCloser(data)
+			}
+			headers["Content-Encoding"] = "gzip"
```
Comment (Ant-hem) [inline algolia/transport/transport.go]: Is the standard http client setting the `Content-Lenght` header as well?

**Comment by Ant-hem** [comment]
Your CI tests are failing, you could disable the `GZIP` feature on the second test client to make the `accout_client` test passing.

File: algolia/transport/transport.go (hunk 1 above)
Comment (aseure) [reply algolia/transport/transport.go]: Yes indeed. As per the documentation of `http.Request.Header`:
```
// For client requests, certain headers such as Content-Length
// and Connection are automatically written when needed and
// values in Header may be ignored. See the documentation
// for the Request.Write method.
```

**Comment by aseure** [comment]
@Ant-hem I've modified some commits as I was globally modifying the default headers passed to every query :/ Now everything is fixed but the `AccountClient.CopyIndex` of course because one of the testing application is running on a cluster where the GZIP feature is not yet enabled. Waiting for this before merging. Feel free to approve once you're fine with the PR.

**Pull request merged**